backend/test/telemetry_service_impl_test.cpp

```diff
@@ -0,0 +1,168 @@
+#include <future>
+#include <gmock/gmock.h>
+#include <grpc++/grpc++.h>
+#include <grpc++/server.h>
+#include <grpc++/server_builder.h>
+#include <memory>
+#include <vector>
+
+#include "telemetry/mocks/telemetry_mock.h"
+#include "telemetry/telemetry_service_impl.h"
+
+namespace {
+
+using testing::_;
+using testing::NiceMock;
+
+using MockTelemetry = NiceMock<dronecore::testing::MockTelemetry>;
+using TelemetryServiceImpl = dronecore::backend::TelemetryServiceImpl<MockTelemetry>;
+using TelemetryService = dronecore::rpc::telemetry::TelemetryService;
+
+using PositionResponse = dronecore::rpc::telemetry::PositionResponse;
+using Position = dronecore::Telemetry::Position;
+
+class TelemetryServiceImplTest : public ::testing::Test
+{
+protected:
+    virtual void SetUp()
+    {
+        _telemetry = std::unique_ptr<MockTelemetry>(new MockTelemetry());
+        _telemetry_service = std::unique_ptr<TelemetryServiceImpl>(new TelemetryServiceImpl(*_telemetry));
```

Comment:
```
_telemetry = new MockTelemetry();
_telemetry_service = new TelemetryServiceImpl(*_telemetry);
```
this would also do.
Ref: https://docs.microsoft.com/en-us/cpp/cpp/smart-pointers-modern-cpp#uses-for-smart-pointers

Reply:
Oh, I did not know that! I'll have a look, thanks :-)

Reply:
Your proposition doesn't build for me... =/